### ISSUE TITLE
Add shortcode support for alternate font (specifically a fixed pitch font)

### ISSUE THREAD

**Issue by instancezero** [vkbo/novelWriter#1776]
It would be useful to have a shortcode for switching to an alternate font. In my use case, this would be a fixed pitch font to indicate machine-generated text, but the solution might benefit from a more general implementation.

Ideas for implementation:
- this is regular text [fixed]this is fixed pitch[/fixed] and back to regular
- this is regular text [alt]this is fixed pitch[/alt] and back to regular

**Comment by locrianz** [comment]
I would love this as a quick way to switch to and from a font that blacks out what I'm writing, so I can just write and not get caught up in my editing brain.

**Comment by vkbo** [comment]
> I would love this as a quick way to switch to and from a font that blacks out what I'm writing, so I can just write and not get caught up in my editing brain.

I think #1721 would be more suitable for that,

**Comment by peter88213** [comment]
> In my use case, this would be a fixed pitch font to indicate machine-generated text,

Why not use backticks like for inline code in Markdown, which is a common practice? 

Incidentally, as a workaround, you could also use existing formatting that is not needed otherwise and reformat it in LibreOffice after exporting. 
~~If I'm not mistaken, the `ODTTextStyle` instances in the ODT exporter yield LO Writer character styles that you can easily redefine.~~

**Comment by vkbo** [comment]
> Why not use backticks like for inline code in Markdown, which is a common practice?

While support for backticks could be added, allowing an alternative font is a more generalised feature.

Personally, I don't like backticks because they are very awkward to type on many keyboard layouts where the backtick key is expected to be followed by a second key for use as an accent. (You have to press backtick twice, or backtick+space to get an actual backtick.)

This aside, I am considering a more general feature for creating text blocks. novelWriter has a syntax for block indent, which I think solves the most common use cases where a the text includes a letter, or text from a sign or poster. But allowing blocks with alternative formats, perhaps a way to map to ODT/OOXML styles, would be useful for "advanced users"? Then the formatting itself could be handled in the office document, and wouldn't clutter the build settings.

Anyway, as for the inline feature request, both support for backtick and `[pre][/pre]` could be useful. They're both fairly standard. Adding new shortcodes is preferable from a performance point of view since adding Markdown codes require an additional entry to process for the syntax highlighter.

**Comment by peter88213** [comment]
Whatever the markup looks like, here are some thoughts on exporting:

Since `ODTTextStyle` can combine different formats to handle nesting and overlapping, I guess that ODF "automatic styles" are generated in *content.xml* that the user cannot redefine. 

For the required use case, a named style (may it be a character style or a paragraph style) should be defined in "styles.xml". Thus the user can redefine the style, or apply his custom document template.

**Comment by vkbo** [comment]
Yeah, novelWriter generates standard named styles as seen in new documents generated by LibreOffice and MS Word. I also add a couple of custom styles. Automatic P and T styles are generated for ODT in the same manner LibreOffice does it. OOXML doesn't have those.

I was thinking maybe add a shortcode that can take a value, where the value is exported as a named style for ODT and DOCX, and a class for HTML. It should be pretty simple to do, and since the parsing is at build time, there is next to now overhead in the editor. Some of them could map to some predefined styles like "Code" and "Block Quote" or whatever. I've been trying to find a way to add the latter.

**Comment by peter88213** [comment]
> I was thinking maybe add a shortcode that can take a value, where the value is exported as a named style

That's an excellent idea, as this could also apply to foreign language passages etc. A universal problem solver!

Update: 
~~However, it should be made clear that inline character styles cannot be combined with other formatting.~~

Second update:
Now I'm curious and have tried it out with OpenOffice: It actually supports nested combinations of named styles and automatic styles. So my statement above is incorrect.

Third update:
Since this also concerns my own application, I experimented a little further and discovered that OpenOffice also supports combinations of named character styles that cannot be set via the *Writer* user interface.

**Comment by vkbo** [comment]
It's also one of those thing you can pull out of the hat when you really need it, and be a very flexible tool/feature. When you don't need it, it is out of sight and doesn't complicate your project. That's sort of the fundamental idea behind the shortcode and why I added them. They can do the complicated thing if and when you need them.
